File: Lecture Demos/Week 4/01ProceduresFunctions/sale_manager.py
sales = []
costPerSale = 10

def addSale(x):
    sales.add(x)

def isProfitable():
    gross = 0
    for sale in sales:
        gross += sale
    loss = len(sales) * costPerSale
    isProfitable = gross > loss
    return isProfitable

def makeProfitable():
    costPerSale -= 1

File: Lecture Demos (Spring 2024)/Week 4/01ProceduresFunctions/SaleManager.java
import java.util.LinkedList;

public class SaleManager {
    LinkedList<Integer> sales; 
    static int costPerSale = 10;

    public SaleManager() {
        sales = new LinkedList<Integer>();
    }

    public void sale(int x) {
        sales.add(x);
    }

    public boolean profitable() {
        int gross = 0;
        for (int sale : sales) {
            gross += sale;
        }
        int loss = sales.size() * costPerSale;
        boolean isProfitable = gross > loss;
        if (!isProfitable) {
            costPerSale -= 1;
        }
        return isProfitable;
    }

    public static void main(String[] args) {
        
    }
}